thread/src/main/kotlin/io/github/paulgriffith/kindling/thread/ThreadView.kt

```diff
@@ -63,18 +65,81 @@ class ThreadView(
                 null,
             ),
         )
-        componentPopupMenu = FlatPopupMenu().apply {
-            add(
-                JMenu("Mark all with same...").apply {
-                    ThreadModel.ThreadColumns.forEach {col ->
-                        add(
-                            Action(col.header) {
-                                markAllWithSameValue(col)
-                            }
-                        )
+
+        fun markAllWithSameValue(property: Column<Thread, *>) {
+            val selectedPropertyValues = selectedRowIndices()
+                .map { index -> model[index, property] }
+                .filterNotNull()
+                .toSet()
+
+            for (row in 0 until model.rowCount) {
+                val currentRowValue = model[row, property]
+                if (currentRowValue in selectedPropertyValues) {
+                    model.setValueAt(true, row, ThreadModel[ThreadModel.Mark])
+                }
+            }
+        }
+
+        fun filterAllWithSameValue(property: Column<Thread, *>) {
+            val firstRow = selectedRowIndices().first()
+            when (property) {
+                ThreadModel.State -> {
+                    val state = model[firstRow, ThreadModel.State]
+                    stateList.select(state)
+                }
+
+                ThreadModel.System -> {
+                    val system = model[firstRow, ThreadModel.System]
+                    if (system != null) {
+                        systemList.select(system)
                     }
                 }
-            )
+
+                ThreadModel.Pool -> {
+                    val pool = model[firstRow, ThreadModel.Pool]
+                    if (pool != null) {
+                        poolList.select(pool)
+                    }
+                }
+            }
+        }
+
+        actionMap.put(
+            "${ColumnControlButton.COLUMN_CONTROL_MARKER}.clearAllMarks",
+            Action(name = "Clear All Marks") {
+                for (thread in model.threads) {
+                    thread.marked = false
+                }
+            },
+        )
+
+        attachPopupMenu { event ->
+            val rowAtPoint = rowAtPoint(event.point)
+            selectionModel.setSelectionInterval(rowAtPoint, rowAtPoint)
```

Comment:
The selection interval is always going to have a size of 1, but `markAllWithSameValue()` does extra work to get a filtered set. Originally this was working for any number of threads selected. Something to consider, do these right-click options need to work for multiple selections?